### PR TITLE
add pre-commit config

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.2.3
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: fix-encoding-pragma
+        args: [--remove]
+    -   id: check-yaml
+    -   id: debug-statements
+        exclude: _pytest/debugging.py
```
Comment (graingert) [inline .pre-commit-config.yaml]:
```suggestion
```

File: pytest_timeout.py [hunk 2]
```diff
@@ -39,58 +38,61 @@
...
-    group.addoption('--timeout-method',
-                    dest='timeout_method',
-                    action='store',
-                    choices=['signal', 'thread'],
-                    help=METHOD_DESC)
-    parser.addini('timeout', TIMEOUT_DESC)
-    parser.addini('timeout_method', METHOD_DESC)
-    parser.addini('timeout_func_only', FUNC_ONLY_DESC, type='bool')
+        "timeout",
+        "Interrupt test run and dump stacks of all threads after " "a test times out",
```
Comment (flub) [inline pytest_timeout.py]: I assume this is black annoying old people by completely ignoring 79 char limits... but that's a pretty poor version of this.  Could you turn this back into a single string at least?

File: pytest_timeout.py [hunk 3]
```diff
@@ -39,58 +38,61 @@
...
 
 
-
 @pytest.hookimpl(hookwrapper=True)
 def pytest_runtest_protocol(item):
     func_only = get_func_only_setting(item)
     if func_only is False:
         timeout_setup(item)
-    outcome = yield
+    yield
```
Comment (flub) [inline pytest_timeout.py]: Is this black doing this?  Does it respect a rustfmt-like convention of using `_outcome` for unused variables?  I do rather like having that variable assignment as it gives a lot of clues to the reader.

File: pytest_timeout.py [hunk 4]
```diff
@@ -100,18 +102,22 @@ def pytest_runtest_call(item):
     func_only = get_func_only_setting(item)
     if func_only is True:
         timeout_setup(item)
-    outcome = yield
+    yield
```
Comment (flub) [inline pytest_timeout.py]: same

File: tox.ini [hunk 5]
```diff
@@ -9,3 +9,16 @@ envlist = py27,py35,py36,pypy2,pypy3
...
+basepython = python3
+deps = pre-commit>=1.11.0
+commands = pre-commit run --all-files --show-diff-on-failure
+
+
+[flake8]
+disable-noqa = True
+max-line-length = 88
+extend-ignore =
+   E203  # whitespace before : is not PEP8 compliant (& conflicts with black)
```
Comment (flub) [inline tox.ini]: oh black

File: pytest_timeout.py (hunk 3 above)
Comment (graingert) [reply pytest_timeout.py]: This wasn't black, it was flake8

**Comment by flub** [comment]
Is there something in pre-commit like black for yaml files?

File: pytest_timeout.py (hunk 2 above)
Comment (graingert) [reply pytest_timeout.py]:
```suggestion
        "Interrupt test run and dump stacks of all threads after a test times out",
```

**Comment by flub** [comment]
Oh, lastly would you mind squashing the commits before merging?

File: pytest_timeout.py (hunk 3 above)
Comment (flub) [reply pytest_timeout.py]: sure, not worth having an argument over.  if the flake8 gods have decided so be it

**Comment by graingert** [comment]
> Excuse my moaning about black's poor choices (double quotes? >79 lines? non-pep8? tssk!), I appreciate the benefit of this. Would be nice if you could address the few comments though.

E203 is non-pep8: https://github.com/PyCQA/pycodestyle/issues/373#issuecomment-393095146

**Comment by graingert** [comment]
> Is there something in pre-commit like black for yaml files?

yes: https://github.com/adrienverge/yamllint
I just went for what's in pytest-dev/pytest -mypy and with py2 compat

**Pull request merged**